base_layer/core/src/validation/error.rs

```diff
@@ -39,11 +42,15 @@ use crate::{
 #[derive(Debug, Error)]
 pub enum ValidationError {
     #[error("Block header validation failed: {0}")]
+    #[cfg(feature = "base_node")]
     BlockHeaderError(#[from] BlockHeaderValidationError),
```

Comment:
There are probably a number of ways to allow this error in multiple features. 
It doesn't feel good to have enum variants different in different builds. This isn't serialized, but it could cause all sorts of errors if it were. 
I'd suggest either using a `dyn Error` or a string or anyhow or something other than the error type if that's what's causing the feature here. 

Otherwise move the error to a crate that can be referenced